{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "1f8d307d-9362-4bc0-a53f-df43e23d5db9",
   "metadata": {},
   "source": [
    "## Ingestão de dados no Banco"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3f5359ef-0a99-4f47-bbf0-a28b51dd4096",
   "metadata": {},
   "source": [
    "### Install dependencias "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "f6850971-c0d2-48d4-b4bd-c306e496e361",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[sudo] password for jovyan: \n"
     ]
    }
   ],
   "source": [
    "!sudo apt-get install libmysqlclient-dev --yes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7dbd60f6-84a8-4f29-9536-08c77032e4d9",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: dataset in /opt/conda/lib/python3.9/site-packages (1.5.1)\n",
      "Collecting console-logging\n",
      "  Downloading console_logging-0.0.5.0-py2.py3-none-any.whl (5.3 kB)\n",
      "Collecting mysqlclient\n",
      "  Downloading mysqlclient-2.1.0.tar.gz (87 kB)\n",
      "     |████████████████████████████████| 87 kB 7.0 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_05a648a9a3ce401997f0ba14492723b0/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_05a648a9a3ce401997f0ba14492723b0/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-iv_st5rt\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_05a648a9a3ce401997f0ba14492723b0/\n",
      "  Complete output (15 lines):\n",
      "  mysql_config --version\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  mariadb_config --version\n",
      "  /bin/sh: 1: mariadb_config: not found\n",
      "  mysql_config --libs\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_05a648a9a3ce401997f0ba14492723b0/setup.py\", line 15, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_05a648a9a3ce401997f0ba14492723b0/setup_posix.py\", line 70, in get_config\n",
      "      libs = mysql_config(\"libs\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_05a648a9a3ce401997f0ba14492723b0/setup_posix.py\", line 31, in mysql_config\n",
      "      raise OSError(\"{} not found\".format(_mysql_config_path))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/de/79/d02be3cb942afda6c99ca207858847572e38146eb73a7c4bfe3bdf154626/mysqlclient-2.1.0.tar.gz#sha256=973235686f1b720536d417bf0a0d39b4ab3d5086b2b6ad5e6752393428c02b12 (from https://pypi.org/simple/mysqlclient/) (requires-python:>=3.5). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-2.0.3.tar.gz (88 kB)\n",
      "     |████████████████████████████████| 88 kB 8.6 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_347cc059c78d43129152962b4d89a01d/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_347cc059c78d43129152962b4d89a01d/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-ia8rravb\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_347cc059c78d43129152962b4d89a01d/\n",
      "  Complete output (15 lines):\n",
      "  mysql_config --version\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  mariadb_config --version\n",
      "  /bin/sh: 1: mariadb_config: not found\n",
      "  mysql_config --libs\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_347cc059c78d43129152962b4d89a01d/setup.py\", line 15, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_347cc059c78d43129152962b4d89a01d/setup_posix.py\", line 70, in get_config\n",
      "      libs = mysql_config(\"libs\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_347cc059c78d43129152962b4d89a01d/setup_posix.py\", line 31, in mysql_config\n",
      "      raise OSError(\"{} not found\".format(_mysql_config_path))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/3c/df/59cd2fa5e48d0804d213bdcb1acb4d08c403b61c7ff7ed4dd4a6a2deb3f7/mysqlclient-2.0.3.tar.gz#sha256=f6ebea7c008f155baeefe16c56cd3ee6239f7a5a9ae42396c2f1860f08a7c432 (from https://pypi.org/simple/mysqlclient/) (requires-python:>=3.5). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-2.0.2.tar.gz (88 kB)\n",
      "     |████████████████████████████████| 88 kB 8.3 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_033d125269ff42c3b1e4c37f52e45700/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_033d125269ff42c3b1e4c37f52e45700/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-_x3injfm\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_033d125269ff42c3b1e4c37f52e45700/\n",
      "  Complete output (12 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  /bin/sh: 1: mariadb_config: not found\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_033d125269ff42c3b1e4c37f52e45700/setup.py\", line 15, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_033d125269ff42c3b1e4c37f52e45700/setup_posix.py\", line 65, in get_config\n",
      "      libs = mysql_config(\"libs\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_033d125269ff42c3b1e4c37f52e45700/setup_posix.py\", line 31, in mysql_config\n",
      "      raise OSError(\"{} not found\".format(_mysql_config_path))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/0e/68/f79de0e0fcdb041783124b825771e0ba6a3d0893b7b4c4735f130e42af47/mysqlclient-2.0.2.tar.gz#sha256=8df057b08fc27d8f7106bfa997d0a21e2acef017f905f06d6fb0aa6a20d4d2b2 (from https://pypi.org/simple/mysqlclient/) (requires-python:>=3.5). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-2.0.1.tar.gz (87 kB)\n",
      "     |████████████████████████████████| 87 kB 8.6 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_7fbdf00d3d0a4d40a06593dcff4fed9b/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_7fbdf00d3d0a4d40a06593dcff4fed9b/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-1ssun8r0\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_7fbdf00d3d0a4d40a06593dcff4fed9b/\n",
      "  Complete output (12 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  /bin/sh: 1: mariadb_config: not found\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_7fbdf00d3d0a4d40a06593dcff4fed9b/setup.py\", line 15, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_7fbdf00d3d0a4d40a06593dcff4fed9b/setup_posix.py\", line 65, in get_config\n",
      "      libs = mysql_config(\"libs\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_7fbdf00d3d0a4d40a06593dcff4fed9b/setup_posix.py\", line 31, in mysql_config\n",
      "      raise OSError(\"{} not found\".format(_mysql_config_path))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/a5/e1/e5f2b231c05dc51d9d87fa5066f90d1405345c54b14b0b11a1c859020f21/mysqlclient-2.0.1.tar.gz#sha256=fb2f75aea14722390d2d8ddf384ad99da708c707a96656210a7be8af20a2c5e5 (from https://pypi.org/simple/mysqlclient/) (requires-python:>=3.5). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-2.0.0.tar.gz (87 kB)\n",
      "     |████████████████████████████████| 87 kB 7.7 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_bef4edf7d1334b1f81f0c3ccbfb90359/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_bef4edf7d1334b1f81f0c3ccbfb90359/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-kv1a9civ\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_bef4edf7d1334b1f81f0c3ccbfb90359/\n",
      "  Complete output (12 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  /bin/sh: 1: mariadb_config: not found\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_bef4edf7d1334b1f81f0c3ccbfb90359/setup.py\", line 15, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_bef4edf7d1334b1f81f0c3ccbfb90359/setup_posix.py\", line 65, in get_config\n",
      "      libs = mysql_config(\"libs\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_bef4edf7d1334b1f81f0c3ccbfb90359/setup_posix.py\", line 31, in mysql_config\n",
      "      raise OSError(\"{} not found\".format(_mysql_config_path))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/ca/cf/dd8124bf8b3d2ae465db9c8ce35c51650c31784b095f41faf6f8cad987a0/mysqlclient-2.0.0.tar.gz#sha256=201dfcc2a3c8a0859ea399947f1af8abe74b706beba0ebac149180d9d7b0f6e8 (from https://pypi.org/simple/mysqlclient/) (requires-python:>=3.5). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-1.4.6.tar.gz (85 kB)\n",
      "     |████████████████████████████████| 85 kB 7.4 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_703b7ec5194e4b6c89d4e22bb1a18e77/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_703b7ec5194e4b6c89d4e22bb1a18e77/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-giqkr9wp\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_703b7ec5194e4b6c89d4e22bb1a18e77/\n",
      "  Complete output (12 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  /bin/sh: 1: mariadb_config: not found\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_703b7ec5194e4b6c89d4e22bb1a18e77/setup.py\", line 16, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_703b7ec5194e4b6c89d4e22bb1a18e77/setup_posix.py\", line 61, in get_config\n",
      "      libs = mysql_config(\"libs\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_703b7ec5194e4b6c89d4e22bb1a18e77/setup_posix.py\", line 29, in mysql_config\n",
      "      raise EnvironmentError(\"%s not found\" % (_mysql_config_path,))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/d0/97/7326248ac8d5049968bf4ec708a5d3d4806e412a42e74160d7f266a3e03a/mysqlclient-1.4.6.tar.gz#sha256=f3fdaa9a38752a3b214a6fe79d7cae3653731a53e577821f9187e67cbecb2e16 (from https://pypi.org/simple/mysqlclient/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-1.4.5.tar.gz (86 kB)\n",
      "     |████████████████████████████████| 86 kB 8.3 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_07eba6ee25d24d90abecd275ed3c2974/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_07eba6ee25d24d90abecd275ed3c2974/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-ab6b7vt2\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_07eba6ee25d24d90abecd275ed3c2974/\n",
      "  Complete output (12 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  /bin/sh: 1: mariadb_config: not found\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_07eba6ee25d24d90abecd275ed3c2974/setup.py\", line 16, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_07eba6ee25d24d90abecd275ed3c2974/setup_posix.py\", line 61, in get_config\n",
      "      libs = mysql_config(\"libs\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_07eba6ee25d24d90abecd275ed3c2974/setup_posix.py\", line 29, in mysql_config\n",
      "      raise EnvironmentError(\"%s not found\" % (_mysql_config_path,))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/f8/9b/5db9a03e2088a87c26e3e4d4c7f7e8f4c2dbae610f9521cdfac15755a795/mysqlclient-1.4.5.tar.gz#sha256=e80109b0ae8d952b900b31b623181532e5e89376d707dcbeb63f99e69cefe559 (from https://pypi.org/simple/mysqlclient/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-1.4.4.tar.gz (86 kB)\n",
      "     |████████████████████████████████| 86 kB 7.3 MB/s            \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_f87fd4212af9405bac5206973cb4f40a/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_f87fd4212af9405bac5206973cb4f40a/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-t8219_sz\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_f87fd4212af9405bac5206973cb4f40a/\n",
      "  Complete output (12 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  /bin/sh: 1: mariadb_config: not found\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_f87fd4212af9405bac5206973cb4f40a/setup.py\", line 16, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_f87fd4212af9405bac5206973cb4f40a/setup_posix.py\", line 61, in get_config\n",
      "      libs = mysql_config(\"libs\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_f87fd4212af9405bac5206973cb4f40a/setup_posix.py\", line 29, in mysql_config\n",
      "      raise EnvironmentError(\"%s not found\" % (_mysql_config_path,))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/4d/38/c5f8bac9c50f3042c8f05615f84206f77f03db79781db841898fde1bb284/mysqlclient-1.4.4.tar.gz#sha256=9c737cc55a5dc8dd3583a942d5a9b21be58d16f00f5fefca4e575e7d9682e98c (from https://pypi.org/simple/mysqlclient/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-1.4.3.tar.gz (86 kB)\n",
      "     |████████████████████████████████| 86 kB 8.7 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_1bf363903af043edb7922e90916e3973/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_1bf363903af043edb7922e90916e3973/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-a4am5vrz\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_1bf363903af043edb7922e90916e3973/\n",
      "  Complete output (12 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  /bin/sh: 1: mariadb_config: not found\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_1bf363903af043edb7922e90916e3973/setup.py\", line 16, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_1bf363903af043edb7922e90916e3973/setup_posix.py\", line 61, in get_config\n",
      "      libs = mysql_config(\"libs\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_1bf363903af043edb7922e90916e3973/setup_posix.py\", line 29, in mysql_config\n",
      "      raise EnvironmentError(\"%s not found\" % (_mysql_config_path,))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/13/37/6c941aabeb78464a538b7d43266bfae645993779e3ca3e12fe82b4590f52/mysqlclient-1.4.3.tar.gz#sha256=0d3eeb97534c9f42e9b65bd35c1a35fadc2f6d7ea5d9fdbda5339a569156344b (from https://pypi.org/simple/mysqlclient/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-1.4.2.post1.tar.gz (85 kB)\n",
      "     |████████████████████████████████| 85 kB 8.3 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_7a616caa1a6e42d488ab9e450cfcedb8/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_7a616caa1a6e42d488ab9e450cfcedb8/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-8u76t_8_\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_7a616caa1a6e42d488ab9e450cfcedb8/\n",
      "  Complete output (10 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_7a616caa1a6e42d488ab9e450cfcedb8/setup.py\", line 16, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_7a616caa1a6e42d488ab9e450cfcedb8/setup_posix.py\", line 51, in get_config\n",
      "      libs = mysql_config(\"libs\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_7a616caa1a6e42d488ab9e450cfcedb8/setup_posix.py\", line 29, in mysql_config\n",
      "      raise EnvironmentError(\"%s not found\" % (_mysql_config_path,))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/f4/f1/3bb6f64ca7a429729413e6556b7ba5976df06019a5245a43d36032f1061e/mysqlclient-1.4.2.post1.tar.gz#sha256=f257d250f2675d0ef99bd318906f3cfc05cef4a2f385ea695ff32a3f04b9f9a7 (from https://pypi.org/simple/mysqlclient/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-1.4.2.tar.gz (85 kB)\n",
      "     |████████████████████████████████| 85 kB 8.3 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_aac484b8f0ae4342818d4b75c2c20595/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_aac484b8f0ae4342818d4b75c2c20595/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-owd26f7f\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_aac484b8f0ae4342818d4b75c2c20595/\n",
      "  Complete output (10 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_aac484b8f0ae4342818d4b75c2c20595/setup.py\", line 16, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_aac484b8f0ae4342818d4b75c2c20595/setup_posix.py\", line 51, in get_config\n",
      "      libs = mysql_config(\"libs\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_aac484b8f0ae4342818d4b75c2c20595/setup_posix.py\", line 29, in mysql_config\n",
      "      raise EnvironmentError(\"%s not found\" % (_mysql_config_path,))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/73/79/67ddf24ac31b05d741f0ac87fa612c7a11bab1b39b8270ed7344d149f8af/mysqlclient-1.4.2.tar.gz#sha256=b95edaa41d6cc47deecabcdcbb5ab437ad9ae6d8955f5cf10d1847b37e66ef5e (from https://pypi.org/simple/mysqlclient/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-1.4.1.tar.gz (85 kB)\n",
      "     |████████████████████████████████| 85 kB 3.3 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_313e9c58efa9473290c8bd372a73650c/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_313e9c58efa9473290c8bd372a73650c/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-0sm4pe5s\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_313e9c58efa9473290c8bd372a73650c/\n",
      "  Complete output (10 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_313e9c58efa9473290c8bd372a73650c/setup.py\", line 16, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_313e9c58efa9473290c8bd372a73650c/setup_posix.py\", line 51, in get_config\n",
      "      libs = mysql_config(\"libs\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_313e9c58efa9473290c8bd372a73650c/setup_posix.py\", line 29, in mysql_config\n",
      "      raise EnvironmentError(\"%s not found\" % (_mysql_config_path,))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/de/d7/919042fb7ed78a89e21c4e43d5daed90d5df18ec183bae3de493a7d9e3ed/mysqlclient-1.4.1.tar.gz#sha256=a62220410e26ce2d2ff94dd0138c3ecfb91db634464a9afb4c8e6b50f0a67e00 (from https://pypi.org/simple/mysqlclient/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-1.4.0.tar.gz (85 kB)\n",
      "     |████████████████████████████████| 85 kB 8.0 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_269b6b1036e945dfa866b3ef1e031768/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_269b6b1036e945dfa866b3ef1e031768/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-v76tyk8m\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_269b6b1036e945dfa866b3ef1e031768/\n",
      "  Complete output (10 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_269b6b1036e945dfa866b3ef1e031768/setup.py\", line 16, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_269b6b1036e945dfa866b3ef1e031768/setup_posix.py\", line 51, in get_config\n",
      "      libs = mysql_config(\"libs\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_269b6b1036e945dfa866b3ef1e031768/setup_posix.py\", line 29, in mysql_config\n",
      "      raise EnvironmentError(\"%s not found\" % (_mysql_config_path,))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/05/73/fac741eef89b0257fb1807de80e96474f40d52739cdbc32c90c4a793e79d/mysqlclient-1.4.0.tar.gz#sha256=efeaacd04feea7d44fef2a5adf4df335b438458f8b89144307dc5a90025761a5 (from https://pypi.org/simple/mysqlclient/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-1.3.14.tar.gz (91 kB)\n",
      "     |████████████████████████████████| 91 kB 6.6 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_57a220508a0f43c085e71d60d7ba0310/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_57a220508a0f43c085e71d60d7ba0310/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-7v8xhjwo\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_57a220508a0f43c085e71d60d7ba0310/\n",
      "  Complete output (10 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_57a220508a0f43c085e71d60d7ba0310/setup.py\", line 16, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_57a220508a0f43c085e71d60d7ba0310/setup_posix.py\", line 53, in get_config\n",
      "      libs = mysql_config(\"libs_r\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_57a220508a0f43c085e71d60d7ba0310/setup_posix.py\", line 28, in mysql_config\n",
      "      raise EnvironmentError(\"%s not found\" % (mysql_config.path,))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/f7/a2/1230ebbb4b91f42ad6b646e59eb8855559817ad5505d81c1ca2b5a216040/mysqlclient-1.3.14.tar.gz#sha256=3981ae9ce545901a36a8b7aed76ed02960a429f75dc53b7ad77fb2f9ab7cd56b (from https://pypi.org/simple/mysqlclient/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-1.3.13.tar.gz (90 kB)\n",
      "     |████████████████████████████████| 90 kB 7.8 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_22fdb3c0aba446e99ee3a957375c7f06/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_22fdb3c0aba446e99ee3a957375c7f06/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-oi81tl0c\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_22fdb3c0aba446e99ee3a957375c7f06/\n",
      "  Complete output (10 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_22fdb3c0aba446e99ee3a957375c7f06/setup.py\", line 18, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_22fdb3c0aba446e99ee3a957375c7f06/setup_posix.py\", line 53, in get_config\n",
      "      libs = mysql_config(\"libs_r\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_22fdb3c0aba446e99ee3a957375c7f06/setup_posix.py\", line 28, in mysql_config\n",
      "      raise EnvironmentError(\"%s not found\" % (mysql_config.path,))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/ec/fd/83329b9d3e14f7344d1cb31f128e6dbba70c5975c9e57896815dbb1988ad/mysqlclient-1.3.13.tar.gz#sha256=ff8ee1be84215e6c30a746b728c41eb0701a46ca76e343af445b35ce6250644f (from https://pypi.org/simple/mysqlclient/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-1.3.12.tar.gz (89 kB)\n",
      "     |████████████████████████████████| 89 kB 6.9 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_829e2e2d84dc44158abf30f8b10e84bd/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_829e2e2d84dc44158abf30f8b10e84bd/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-ygbkc_1l\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_829e2e2d84dc44158abf30f8b10e84bd/\n",
      "  Complete output (10 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_829e2e2d84dc44158abf30f8b10e84bd/setup.py\", line 17, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_829e2e2d84dc44158abf30f8b10e84bd/setup_posix.py\", line 44, in get_config\n",
      "      libs = mysql_config(\"libs_r\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_829e2e2d84dc44158abf30f8b10e84bd/setup_posix.py\", line 26, in mysql_config\n",
      "      raise EnvironmentError(\"%s not found\" % (mysql_config.path,))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/6f/86/bad31f1c1bb0cc99e88ca2adb7cb5c71f7a6540c1bb001480513de76a931/mysqlclient-1.3.12.tar.gz#sha256=2d9ec33de39f4d9c64ad7322ede0521d85829ce36a76f9dd3d6ab76a9c8648e5 (from https://pypi.org/simple/mysqlclient/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-1.3.11.tar.gz (89 kB)\n",
      "     |████████████████████████████████| 89 kB 8.2 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_a92395ecb6f340e5a72050d786206ae9/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_a92395ecb6f340e5a72050d786206ae9/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-o9xrlkeo\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_a92395ecb6f340e5a72050d786206ae9/\n",
      "  Complete output (10 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_a92395ecb6f340e5a72050d786206ae9/setup.py\", line 17, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_a92395ecb6f340e5a72050d786206ae9/setup_posix.py\", line 44, in get_config\n",
      "      libs = mysql_config(\"libs_r\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_a92395ecb6f340e5a72050d786206ae9/setup_posix.py\", line 26, in mysql_config\n",
      "      raise EnvironmentError(\"%s not found\" % (mysql_config.path,))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/6a/4d/856d1848e9b7fa8577a31c0188592f472e5cbc94f588c2a40ff930533d73/mysqlclient-1.3.11.tar.gz#sha256=9a79b38843d61f651266ff16b9fdcc8830d21f0fb852e4406a19e9a715bdf3e2 (from https://pypi.org/simple/mysqlclient/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-1.3.10.tar.gz (82 kB)\n",
      "     |████████████████████████████████| 82 kB 5.1 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_c326dea6e67c4601a863890483a5e0be/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_c326dea6e67c4601a863890483a5e0be/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-ttrtm4bx\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_c326dea6e67c4601a863890483a5e0be/\n",
      "  Complete output (10 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_c326dea6e67c4601a863890483a5e0be/setup.py\", line 17, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_c326dea6e67c4601a863890483a5e0be/setup_posix.py\", line 44, in get_config\n",
      "      libs = mysql_config(\"libs_r\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_c326dea6e67c4601a863890483a5e0be/setup_posix.py\", line 26, in mysql_config\n",
      "      raise EnvironmentError(\"%s not found\" % (mysql_config.path,))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/40/9b/0bc869f290b8f49a99b8d97927f57126a5d1befcf8bac92c60dc855f2523/mysqlclient-1.3.10.tar.gz#sha256=452aaf9d2ba1a4fb85c134c88f9b1857852deed63b3b13255d7a65d2c1297262 (from https://pypi.org/simple/mysqlclient/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-1.3.9.tar.gz (81 kB)\n",
      "     |████████████████████████████████| 81 kB 7.3 MB/s            \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_381ae5e988554d088d7633e5076f4502/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_381ae5e988554d088d7633e5076f4502/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-wjija_61\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_381ae5e988554d088d7633e5076f4502/\n",
      "  Complete output (10 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_381ae5e988554d088d7633e5076f4502/setup.py\", line 17, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_381ae5e988554d088d7633e5076f4502/setup_posix.py\", line 44, in get_config\n",
      "      libs = mysql_config(\"libs_r\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_381ae5e988554d088d7633e5076f4502/setup_posix.py\", line 26, in mysql_config\n",
      "      raise EnvironmentError(\"%s not found\" % (mysql_config.path,))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/db/f5/c8e1657985c31dda16e434edf5257c31572fa5faacd7e48b1618390e4b18/mysqlclient-1.3.9.tar.gz#sha256=990ccf1e1f15b9a291b811c993dc1c216566714bc14e6581b38665bd61c28c99 (from https://pypi.org/simple/mysqlclient/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-1.3.8.tar.gz (81 kB)\n",
      "     |████████████████████████████████| 81 kB 9.1 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_83ec8aa79a824e9e83d1b7383e710a95/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_83ec8aa79a824e9e83d1b7383e710a95/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-jdlp_046\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_83ec8aa79a824e9e83d1b7383e710a95/\n",
      "  Complete output (10 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_83ec8aa79a824e9e83d1b7383e710a95/setup.py\", line 17, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_83ec8aa79a824e9e83d1b7383e710a95/setup_posix.py\", line 44, in get_config\n",
      "      libs = mysql_config(\"libs_r\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_83ec8aa79a824e9e83d1b7383e710a95/setup_posix.py\", line 26, in mysql_config\n",
      "      raise EnvironmentError(\"%s not found\" % (mysql_config.path,))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/31/ce/7ea049b3d5929b3cce3104967ccca218a9f054517f0d15dfdaf4e76da2a0/mysqlclient-1.3.8.tar.gz#sha256=c64bc386c0b771ed6ff76de0f52e0f957f98d9e13c83b0ddb821afdbe3f5d970 (from https://pypi.org/simple/mysqlclient/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-1.3.7.tar.gz (79 kB)\n",
      "     |████████████████████████████████| 79 kB 8.5 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_1a8740c0968c443e826c6811fd8fbe71/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_1a8740c0968c443e826c6811fd8fbe71/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-r63kyrp9\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_1a8740c0968c443e826c6811fd8fbe71/\n",
      "  Complete output (10 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_1a8740c0968c443e826c6811fd8fbe71/setup.py\", line 17, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_1a8740c0968c443e826c6811fd8fbe71/setup_posix.py\", line 44, in get_config\n",
      "      libs = mysql_config(\"libs_r\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_1a8740c0968c443e826c6811fd8fbe71/setup_posix.py\", line 26, in mysql_config\n",
      "      raise EnvironmentError(\"%s not found\" % (mysql_config.path,))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/74/ff/4e964e20b559e55d7afa60fbccc6a560f2adf289813bd3d7eb4eb8a87093/mysqlclient-1.3.7.tar.gz#sha256=c74a83b4cb2933d0e43370117eeebdfa03077ae72686d2df43d31879267f1f1b (from https://pypi.org/simple/mysqlclient/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-1.3.6.tar.gz (78 kB)\n",
      "     |████████████████████████████████| 78 kB 9.1 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_72047f6517334ed6b5d294417ccdf567/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_72047f6517334ed6b5d294417ccdf567/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-aach668n\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_72047f6517334ed6b5d294417ccdf567/\n",
      "  Complete output (10 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_72047f6517334ed6b5d294417ccdf567/setup.py\", line 17, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_72047f6517334ed6b5d294417ccdf567/setup_posix.py\", line 47, in get_config\n",
      "      libs = mysql_config(\"libs_r\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_72047f6517334ed6b5d294417ccdf567/setup_posix.py\", line 29, in mysql_config\n",
      "      raise EnvironmentError(\"%s not found\" % (mysql_config.path,))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/09/29/6648563af4b45798ba667f17ea55166d7b0ed8717937c06977a1fdbe7df6/mysqlclient-1.3.6.tar.gz#sha256=1f63a8ab111530a0ff9acaeb118074f11e7564b7c124f2e7fc56fcac641aecf2 (from https://pypi.org/simple/mysqlclient/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-1.3.5.tar.gz (77 kB)\n",
      "     |████████████████████████████████| 77 kB 7.8 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_6ae92978f5fb41e4a004335395ee75db/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_6ae92978f5fb41e4a004335395ee75db/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-ypqsem50\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_6ae92978f5fb41e4a004335395ee75db/\n",
      "  Complete output (10 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_6ae92978f5fb41e4a004335395ee75db/setup.py\", line 17, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_6ae92978f5fb41e4a004335395ee75db/setup_posix.py\", line 47, in get_config\n",
      "      libs = mysql_config(\"libs_r\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_6ae92978f5fb41e4a004335395ee75db/setup_posix.py\", line 29, in mysql_config\n",
      "      raise EnvironmentError(\"%s not found\" % (mysql_config.path,))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/82/ea/336daf6c7baa564d0c434ab748218d1a7dc61dfc7066e98b72afd24fd2b9/mysqlclient-1.3.5.tar.gz#sha256=80ea68f6f3c2cfbf35c91adde2b40de1f2c31132118d2dcee048c72423f678bf (from https://pypi.org/simple/mysqlclient/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-1.3.4.tar.gz (77 kB)\n",
      "     |████████████████████████████████| 77 kB 8.9 MB/s            \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_7c37399df37d47178f36f593dcaa6fec/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_7c37399df37d47178f36f593dcaa6fec/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-2p7zaikk\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_7c37399df37d47178f36f593dcaa6fec/\n",
      "  Complete output (10 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_7c37399df37d47178f36f593dcaa6fec/setup.py\", line 17, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_7c37399df37d47178f36f593dcaa6fec/setup_posix.py\", line 47, in get_config\n",
      "      libs = mysql_config(\"libs_r\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_7c37399df37d47178f36f593dcaa6fec/setup_posix.py\", line 29, in mysql_config\n",
      "      raise EnvironmentError(\"%s not found\" % (mysql_config.path,))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/17/d9/03b08e6a033401335b720806b87596aee8d7f7aa87539713238905c4a8d1/mysqlclient-1.3.4.tar.gz#sha256=a0263383f8ac33ad12d68dc61fa8ccbee41244ce916287c6cee71bf8aea82c5c (from https://pypi.org/simple/mysqlclient/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-1.3.3.tar.gz (77 kB)\n",
      "     |████████████████████████████████| 77 kB 2.4 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_a8964949f9f44130a5ad72cf36fb632b/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_a8964949f9f44130a5ad72cf36fb632b/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-s61n1_50\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_a8964949f9f44130a5ad72cf36fb632b/\n",
      "  Complete output (10 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_a8964949f9f44130a5ad72cf36fb632b/setup.py\", line 17, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_a8964949f9f44130a5ad72cf36fb632b/setup_posix.py\", line 47, in get_config\n",
      "      libs = mysql_config(\"libs_r\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_a8964949f9f44130a5ad72cf36fb632b/setup_posix.py\", line 29, in mysql_config\n",
      "      raise EnvironmentError(\"%s not found\" % (mysql_config.path,))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/26/53/c575db342bfca0213c9243ed2642bfadcd420b18d2f477dc812543e3d83a/mysqlclient-1.3.3.tar.gz#sha256=0f0887882d71bf2a079d9724d9323053cfbee8ab2dcfc989daf97d7ce5be9af9 (from https://pypi.org/simple/mysqlclient/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-1.3.2.tar.gz (77 kB)\n",
      "     |████████████████████████████████| 77 kB 8.9 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_6b80cb5b89f44503b8cf3f9c7e0c071d/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_6b80cb5b89f44503b8cf3f9c7e0c071d/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-k7oxdulu\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_6b80cb5b89f44503b8cf3f9c7e0c071d/\n",
      "  Complete output (10 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_6b80cb5b89f44503b8cf3f9c7e0c071d/setup.py\", line 17, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_6b80cb5b89f44503b8cf3f9c7e0c071d/setup_posix.py\", line 47, in get_config\n",
      "      libs = mysql_config(\"libs_r\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_6b80cb5b89f44503b8cf3f9c7e0c071d/setup_posix.py\", line 29, in mysql_config\n",
      "      raise EnvironmentError(\"%s not found\" % (mysql_config.path,))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/2d/df/5440ee86bbb248325cdd6e1fc9cbbba365018a0d2d57f673610e020e6b1d/mysqlclient-1.3.2.tar.gz#sha256=e44025830501b9f70f8c2fe8eeff66f0df2df198802f7295801dac199b8236ef (from https://pypi.org/simple/mysqlclient/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-1.3.1.tar.gz (76 kB)\n",
      "     |████████████████████████████████| 76 kB 5.2 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_2e14821d33424fa385302efc71360276/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_2e14821d33424fa385302efc71360276/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-hyhtj8p3\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_2e14821d33424fa385302efc71360276/\n",
      "  Complete output (10 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_2e14821d33424fa385302efc71360276/setup.py\", line 17, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_2e14821d33424fa385302efc71360276/setup_posix.py\", line 47, in get_config\n",
      "      libs = mysql_config(\"libs_r\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_2e14821d33424fa385302efc71360276/setup_posix.py\", line 29, in mysql_config\n",
      "      raise EnvironmentError(\"%s not found\" % (mysql_config.path,))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/6b/ba/4729d99e85a0a35bb46d55500570de05b4af10431cef174b6da9f58a0e50/mysqlclient-1.3.1.tar.gz#sha256=3549e8a61f10c8cd8eac6581d3f44d0594f535fb7b29e6090db3a0bc547b25ad (from https://pypi.org/simple/mysqlclient/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "  Downloading mysqlclient-1.3.0.tar.gz (76 kB)\n",
      "     |████████████████████████████████| 76 kB 7.7 MB/s             \n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /opt/conda/bin/python3.9 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_d8c88c029c8c4b2c878ad38ae612f9d7/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-x8k336cn/mysqlclient_d8c88c029c8c4b2c878ad38ae612f9d7/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base /tmp/pip-pip-egg-info-j73fa0ad\n",
      "       cwd: /tmp/pip-install-x8k336cn/mysqlclient_d8c88c029c8c4b2c878ad38ae612f9d7/\n",
      "  Complete output (10 lines):\n",
      "  /bin/sh: 1: mysql_config: not found\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_d8c88c029c8c4b2c878ad38ae612f9d7/setup.py\", line 17, in <module>\n",
      "      metadata, options = get_config()\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_d8c88c029c8c4b2c878ad38ae612f9d7/setup_posix.py\", line 47, in get_config\n",
      "      libs = mysql_config(\"libs_r\")\n",
      "    File \"/tmp/pip-install-x8k336cn/mysqlclient_d8c88c029c8c4b2c878ad38ae612f9d7/setup_posix.py\", line 29, in mysql_config\n",
      "      raise EnvironmentError(\"%s not found\" % (mysql_config.path,))\n",
      "  OSError: mysql_config not found\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[?25h\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/6a/91/bdfe808fb5dc99a5f65833b370818161b77ef6d1e19b488e4c146ab615aa/mysqlclient-1.3.0.tar.gz#sha256=06eb5664e3738b283ea2262ee60ed83192e898f019cc7ff251f4d05a564ab3b7 (from https://pypi.org/simple/mysqlclient/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\u001b[0m\n",
      "\u001b[31mERROR: Could not find a version that satisfies the requirement mysqlclient (from versions: 1.3.0, 1.3.1, 1.3.2, 1.3.3, 1.3.4, 1.3.5, 1.3.6, 1.3.7, 1.3.8, 1.3.9, 1.3.10, 1.3.11rc1, 1.3.11, 1.3.12, 1.3.13, 1.3.14, 1.4.0rc1, 1.4.0rc2, 1.4.0rc3, 1.4.0, 1.4.1, 1.4.2, 1.4.2.post1, 1.4.3, 1.4.4, 1.4.5, 1.4.6, 2.0.0, 2.0.1, 2.0.2, 2.0.3, 2.1.0rc1, 2.1.0)\u001b[0m\n",
      "\u001b[31mERROR: No matching distribution found for mysqlclient\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!pip install dataset console-logging mysqlclient dataset --no-cache-dir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "f147d969-bfd2-4914-b977-6bafcf07074d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import dataset\n",
    "\n",
    "db = dataset.connect('mysql://user:password@localhost/mydatabase')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b01d0a58-db1e-4fc6-9cc6-e19190dcdc0d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1f491097-9427-4b39-9cc7-bb70881fac7e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting dataset\n",
      "  Downloading dataset-1.5.1-py2.py3-none-any.whl (18 kB)\n",
      "Collecting banal>=1.0.1\n",
      "  Downloading banal-1.0.6-py2.py3-none-any.whl (6.1 kB)\n",
      "Requirement already satisfied: alembic>=0.6.2 in /opt/conda/lib/python3.9/site-packages (from dataset) (1.7.4)\n",
      "Requirement already satisfied: sqlalchemy>=1.3.2 in /opt/conda/lib/python3.9/site-packages (from dataset) (1.4.26)\n",
      "Requirement already satisfied: Mako in /opt/conda/lib/python3.9/site-packages (from alembic>=0.6.2->dataset) (1.1.5)\n",
      "Requirement already satisfied: greenlet!=0.4.17 in /opt/conda/lib/python3.9/site-packages (from sqlalchemy>=1.3.2->dataset) (1.1.2)\n",
      "Requirement already satisfied: MarkupSafe>=0.9.2 in /opt/conda/lib/python3.9/site-packages (from Mako->alembic>=0.6.2->dataset) (2.0.1)\n",
      "Installing collected packages: banal, dataset\n",
      "Successfully installed banal-1.0.6 dataset-1.5.1\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "346d960e-7cbe-43f1-a4da-ea6f2cf738f7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "OrderedDict([('id', 1), ('name', 'John Doe'), ('age', 37), ('gender', None)])"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "john"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "87542de9-84c3-48b6-930e-205133d667dd",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
